app/src/main/java/com/jetbrains/handson/mpp/mobile/MainActivity.kt

```diff
@@ -8,27 +8,35 @@ import androidx.appcompat.app.AlertDialog
 import androidx.recyclerview.widget.LinearLayoutManager
 import kotlinx.android.synthetic.main.activity_main.*
 import kotlinx.android.synthetic.main.recyclerview_row.*
+import java.lang.Integer.max
+import java.lang.Integer.min
 import java.util.*
 
 class MainActivity : AppCompatActivity(), ApplicationContract.View {
 
     private var stations: List<String> = listOf()
     private lateinit var presenter: ApplicationContract.Presenter
 
+    private val buttonToLabel = mapOf(R.id.adultDec to R.id.adultCounter, R.id.adultInc to R.id.adultCounter, R.id.childDec to R.id.childCounter, R.id.childInc to R.id.childCounter)
+    private val buttonToModifier = mapOf(R.id.adultDec to -1, R.id.adultInc to 1, R.id.childDec to -1, R.id.childInc to 1)
+
+    private val minNumberTickets = 0
+    private val maxNumberTickets = 8
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         presenter = ApplicationPresenter()
         presenter.onViewTaken(this)
 
-        setUpSpinners()
         setUpSubmitButton()
         setUpTimePicker()
         supportActionBar?.setTitle("Train\uD83D\uDE82")
     }
 
-    override fun setStationNames(stationNames: List<String>) {
-        stations = stationNames
+    override fun setStationNames(stationNames: List<StationDetails>) {
+        stations = stationNames.map { it.crs!! }
```

Comment:
I feel that this would be more readable if the names of the `stationNames` and `stations` variables were swapped.